[Bug]: Header file inconsistency detected combining OpenSSL and Encrypted Client Hello

### Contact Details

peter.kovar@reflexion.tv

### Version

master

### Description

Configuration error combining --enable-opensslall and --enable-ech

Fedora Linux 40

### Reproduction steps

I have encountered the configuration error while adding ECH option.

./configure --enable-curve25519 --enable-ed25519 --enable-opensslall --enable-opensslextra --enable-dtls --enable-dtls13 --enable-certgen --enable-keygen --enable-intelasm --enable-benchmark --enable-harden --enable-harden-tls --enable-ech


### Relevant log output

```shell
configure: Generating user options header...
checking for wolfssl/openssl/aes.h... no
configure: error: Header file inconsistency detected -- error including wolfssl/openssl/aes.h.
```

--- a/tests/api.c
+++ b/tests/api.c
@@ -27713,7 +27713,7 @@ static int test_wolfSSL_cert_cb_dyn_ciphers(void)
                 "ECDSA+SHA256", caEccCertFile, wolfTLSv1_3_server_method},
 #endif
 #endif
-#ifndef WOLFSSL_NO_TLS12
+#if !defined(WOLFSSL_NO_TLS12)  && !defined(WOLFSSL_HARDEN_TLS)
 #if !defined(NO_RSA) && defined(WC_RSA_PSS) && !defined(NO_DH)
         {wolfTLSv1_2_client_method,
                 "DHE-RSA-AES128-GCM-SHA256",

